docs/types-and-fields.md

```diff
@@ -54,12 +54,16 @@ For simple types (like `string` or `boolean`) this is all that's needed but due
 
 Why use function syntax and not a simple `{ type: Rate }` config object? Because, by using function syntax we solve the problem of circular dependencies (e.g. Post <--> User), so it was adopted as a convention. You can use the shorthand syntax `@Field(() => Rate)` if you want to save some keystrokes but it might be less readable for others.
 
+For nested arrays however, the explicit `[ ]` notation is required to determine the depth of the array. `@Field(type => [[Int]])` would tell the compiler we expect an integer array of depth 2.
```

Comment:
This should be located paragraph earlier, just after "we have two options available" explanation :)

Reply:
Done